src/settings/cloud.ts

```diff
@@ -0,0 +1,174 @@
+import { createServer } from 'http';
+import { getConfig, getConfigLocation, setConfigBulk } from '../config';
+import { getGoofCordFolderPath } from "../utils";
+import { shell } from 'electron';
+
+import fs from 'fs';
+import path from 'path';
+
+function getCloudConfigLocation(): string {
+    const check = path.join(getGoofCordFolderPath(), "cloud.json");
+    if (!fs.existsSync(check)) {
```

Comment:
Checking if a file exists is a bad practice. Try to open the file in getCloudToken() and if it errors, then create the file

Reply:
https://nodejs.org/api/fs.html#fs_fs_exists_path_callback

> Using fs.exists() to check for the existence of a file before calling fs.open(), fs.readFile(), or fs.writeFile() is not recommended. Doing so introduces a race condition, since other processes may change the file's state between the two calls. Instead, user code should open/read/write the file directly and handle the error raised if the file does not exist.